osrsbox/items_api/all_items.py

```diff
@@ -0,0 +1,118 @@
+"""
+Author:  PH01L
+Email:   phoil@osrsbox.com
+Website: https://www.osrsbox.com
+
+Copyright (c) 2019, PH01L
+
+###############################################################################
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+"""
+
+import os
+import json
+import glob
+from typing import Dict
+from typing import List
+from typing import Generator
+
+from osrsbox.items_api import item_definition
+
+
+class AllItems:
+    """This class handles loading of the osrsbox-db items database.
+
+    :param input_data_file_or_directory: The osrsbox-db items folder of JSON files, or single JSON file.
+    """
+    def __init__(self, input_data_file_or_directory: str):
+        self.all_items: List[object] = list()

```

Comment:
You should use the name of the class that this is going to be a list of instead of `object`. I'm not sure `object` is a valid type to `mypy` as well.